File: 013-MaxExponents/MaxExponents.cpp
#include<iostream>
using namespace std;

int countExponentOf2(int n){
    int Count = 0;
    while(n % 2 == 0){
       Count++;
       n/=2;
    }

    return Count;
}
int MaxExponents(int a, int b){
     int MaxNum = a;
     int MaxExp = countExponentOf2(a);

     for (int i = a+1; i <= b; i++){
        int CurrentExp = countExponentOf2(i);
        if(CurrentExp > MaxExp || (CurrentExp == MaxExp && i < MaxExp)){
            MaxExp = CurrentExp;
            MaxNum = i;
        }
     }
     return MaxNum;
}
int main() {
    int a = 7, b = 12;
    cout << "Number with maximum exponent of 2: " << MaxExponents(a, b) << endl;
    return 0;
}

File: 013-MaxExponents/MaxExponents.py
def count_exponent_of_2(n):
    count = 0
    while n % 2 == 0:
        count += 1
        n //= 2
    return count

# Function to find number with maximum exponent of 2 in the range [a, b]
def MaxExponents(a, b):
    max_num = a
    max_exp = count_exponent_of_2(a)

    for i in range(a + 1, b + 1):
        current_exp = count_exponent_of_2(i)
        if current_exp > max_exp or (current_exp == max_exp and i < max_num):
            max_exp = current_exp
            max_num = i
    return max_num

a = 7
b = 12
result = MaxExponents(a, b)
print("Number with maximum exponent of 2:", result)